runtime/src/main/java/org/corfudb/recovery/FastObjectLoader.java

```diff
@@ -728,20 +719,21 @@ public void loadMaps() {
 
     /**
      * This method will apply for each address the consumer given in parameter.
-     *
      * The Necromancer thread is used to do the heavy lifting.
-     * @param logDataProcessor
      */
     private void applyForEachAddress(BiConsumer<Long, ILogData> logDataProcessor) {
 
         summonNecromancer();
         nextRead = logHead;
         while (nextRead <= logTail) {
             try {
-                final long start = nextRead;
-                final long stopNotIncluded = Math.min(start + batchReadSize, logTail + 1);
-                nextRead = stopNotIncluded;
-                final Map<Long, ILogData> range = getLogData(runtime, start, stopNotIncluded);
+                final long lower = nextRead;
+                final long upper = Math.min(lower + batchReadSize - 1, logTail);
+                nextRead = upper + 1;
+                Map<Long, ILogData> range =
+                        runtime.getAddressSpaceView().fetchAll(ContiguousSet.create(
+                                Range.closed(lower, upper), DiscreteDomain.longs()), true);
+                range = new TreeMap<>(range);
```

Comment:
hmm, don't reuse the variable?
I suggest you created a new variable SortedRange

Reply:
Removed. sorting provided in AddressSpaceView